flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -566,9 +560,9 @@ private BufferOrEvent transformToBufferOrEvent(
 			Buffer buffer,
 			boolean moreAvailable,
 			InputChannel currentChannel) throws IOException, InterruptedException {
-		numBytesIn.inc(buffer.getSizeUnsafe());
+		final int size = buffer.getSizeUnsafe();
```

Comment:
Hmm, it looks like we can drop `getSizeUnsafe()` method altogether, since it doesn't differ from `getSize()` - both are not synchronized (also like `setSize()`). It seems like I or @NicoK missed that when we were implementing low latency changes and dropping the synchronisation in the `Buffer` class. Or am I missing something?

Reply:
Yes, it could be replaced by `getSize` here. I could submit a separate hotfix commit for it?

Reply:
Yes, separate hotfix commit would be best (sorry for nit picking here, just noticed this here, since I got curious whether you should be using unsafe or safe version)